lua/nvim-treesitter/query.lua

```diff
@@ -61,9 +62,22 @@ do
     return cache[lang][query_name]
   end
 
+  --- reloads the query file cache
   function M.reload_file_cache(lang, query_name)
+    if lang then
+      if cache[lang] == nil then
+        cache[lang] = {}
+      end
+    end
+
+    -- nil means query is not cached yet. "invalid" means that query was cached and was not a valid one
     if lang and query_name then
-      cache[lang][query_name] = tsq.get_query(lang, query_name)
+      local got = tsq.get_query(lang, query_name)
+      if got == nil then
+        cache[lang][query_name] = "invalid"
```

Comment:
I am not sure we should do the disctinction here, if there was an error, there is a lot of chances the error will be fixed, then the query will be resourced, so using nil is fine here.